docs/source/contributing/development.md

```diff
@@ -0,0 +1,451 @@
+# Development Guide
+
+## Setup
+
+Setup should be easy, as all our packages are isolated in their own _.venv_ maintained
+by [Poetry](https://python-poetry.org/) and for managing the "meta-venv" itself, we
+have the [invocations](../../../invocations) set of helpers controlled with the
+`inv[oke]` command based on the supported [tasks](../../../tasks.py).
+
+So the first thing to run is `inv -l` to see the available list of commands and if you
+see just one, it means that you need to run `inv install-invocations` once, right
+in the root repo directory. (the _meta_ project)
```

Comment:
You mention the "meta-venv" in the first paragraph but this is the first time you mention  the meta project. If we are going to be going back and forth with terms we should either define them or include a definitions section.